BlackNectar/BlackNectar/SideMenuFilterViewController.swift

```diff
@@ -63,16 +65,19 @@ class SideMenuFilterViewController: UITableViewController {
             slideValueLabel.text = "\(roundedNumber)"
         }
         
+        defaultPreferences.set(distanceFilter, forKey: "distanceFilter")
+        
     }
 
     @IBAction func openNowSwitchOffOn(_ sender: Any) {
 
-        if openNowSwitch.isOn {
+        if openNowSwitch.isOn == true {
 
             isOpenNow = true
 
         } else {
-
+            
+            openNowSwitch.isOn = false
```

Comment:
You don't need to set it to false, because will already be false if this statement hits.